src/app/domain/backup/services/uploadMedias.ts

```diff
@@ -0,0 +1,196 @@
+
+  return new Promise((resolve, reject) => {
+    const options = {
+      url: uploadUrl,
+      path: filepath,
+      method: 'POST',
+      type: 'raw',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': media.type,
+        Authorization: `Bearer ${
+          // @ts-expect-error Type issue which will be fixed in another PR
+          client.getStackClient().token.accessToken as string
+        }`
+      },
+      notification: {
+        enabled: false
+      }
+    } as UploadOptions
+
+    RNBackgroundUpload.startUpload(options)
+      .then(uploadId => {
+        RNBackgroundUpload.addListener('error', uploadId, data => {
+          reject({ success: false, data })
+        })
+        RNBackgroundUpload.addListener('cancelled', uploadId, data => {
+          reject({ success: false, data })
+        })
+        RNBackgroundUpload.addListener('completed', uploadId, data => {
+          if (data.responseCode === 201) {
+            resolve({ success: true, data })
+          } else {
+            reject({ success: false, data })
+          }
+        })
+      })
+      .catch(e => {
+        log.debug(e)
+        reject({ status: false, data: {} })
+      })
+  })
+}
+
+const uploadMediaIOS = async (
+  client: CozyClient,
+  uploadUrl: string,
+  media: Media
+): Promise<UploadMediaResult> => {
+  let filepath: string
+
+  if (media.type === 'image') {
+    const data = await CameraRoll.iosGetImageDataById(media.path)
+
+    if (data.node.image.filepath === null) {
+      return { success: false }
+    }
+
+    filepath = data.node.image.filepath.replace('file://', '')
+  } else {
+    try {
+      await RNFileSystem.unlink(
+        RNFileSystem.TemporaryDirectoryPath + media.name
+      )
+    } catch {
+      // we try to remove temporary file just in case it exists
+    }
+
+    filepath = await RNFileSystem.copyAssetsVideoIOS(
+      media.path,
+      RNFileSystem.TemporaryDirectoryPath + media.name
+    )
+  }
+
+  return new Promise((resolve, reject) => {
+    RNFileSystem.uploadFiles({
+      toUrl: uploadUrl,
+      files: [
+        {
+          name: media.name,
+          filename: media.name,
+          filetype: media.type,
+          filepath
+        }
+      ],
+      binaryStreamOnly: true,
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': media.type,
+        Authorization: `Bearer ${
+          // @ts-expect-error Type issue which will be fixed in another PR
+          client.getStackClient().token.accessToken as string
+        }`
+      }
+    })
+      .promise.then(response => {
+        if (response.statusCode == 201) {
+          resolve({ success: true, data: response })
+        } else {
+          reject({ success: false, data: response })
+        }
+      })
+      .catch(e => {
+        log.debug(e)
+        reject({ status: false, data: {} })
+      })
+  })
+}
+
+const getDriveUrl = (
```

Comment:
I would not call that `getDriveUrl`. `getUploadUrl` `getCozyUploadUrl` or what you want, but there is nothing related to the `Drive` app in this method